velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -242,16 +255,26 @@ void WindowFuzzer::go() {
           generateSortingKeysAndOrders("s", argNames, argTypes);
     }
     const auto partitionKeys = generateSortingKeys("p", argNames, argTypes);
-    const auto [frameClause, isRowsFrame] = generateFrameClause();
-    const auto input =
-        generateInputDataWithRowNumber(argNames, argTypes, signature);
+
+    // Randomly select if ROWS or RANGE frame
+    auto typeOfFrame = vectorFuzzer_.coinToss(0.5)
+        ? core::WindowNode::WindowType::kRows
+        : core::WindowNode::WindowType::kRange;
+
     // If the function is order-dependent or uses "rows" frame, sort all input
     // rows by row_number additionally.
-    if (requireSortedInput || isRowsFrame) {
+    if (requireSortedInput ||
+        typeOfFrame == core::WindowNode::WindowType::kRows) {
       sortingKeysAndOrders.emplace_back("row_number", core::kAscNullsLast);
       ++stats_.numSortedInputs;
     }
 
+    const auto [frameClause, hasKRangeFrameBound] =
+        generateFrameClause(typeOfFrame, sortingKeysAndOrders.size() == 1);
+
+    const auto input = generateInputDataWithRowNumber(
```

Comment:
We might want to generate the input data before setting up the frame clause since frames changes the columns.

Reply:
@aditi-pandit In generateInputDataWithRowNumber(), it creates `FLAG_num_batches` of batches of the input data with all its necessary columns. This is why I added the k0 and k1 columns here so they can be included in each batch... 

If I were to move the adding of k0 and k1 columns to after generateInputDataWithRowNumber() and in generateFrameClause(), did you want me to add the k0 and k1 columns to the already created batches in the input data?

Reply:
@minhancao : We should add k0 and k1 columns to the already created batches.

Reply:
Thanks to @pramodsatya, we have come up with some code to modify the already created batches and add the k0 column in:
```
  auto column = vectorMaker.flatVector<int64_t>(
      size, [offset](auto row) { return row + offset; });

  for (auto j = 0; j < FLAGS_num_batches; ++j) {
    auto rowVector = input[j];
    auto names = asRowType(rowVector->type())->names();
    auto childrenTypes = asRowType(rowVector->type())->children();
    names.push_back("k0");
    childrenTypes.push_back(BIGINT());
    rowVector->children().push_back(column);
  }
```

This worked, however we are unable to increment childrenSize_ variable to reflect the new k0 column added in. Furthermore, the childrenSize_ variable is defined as `const size_t childrenSize_;` for RowVector in ComplexVector.h which is why we can't modify it. 

<img width="631" alt="image" src="https://github.com/facebookincubator/velox/assets/36718441/163969e7-cc1e-4f76-888d-8eea73195f51">


@aditi-pandit Do you know if there is a better way to approach this?

If not, I think it is cleaner to add the k0 and k1 columns while generating the input data instead of when we generate the frame clause...

Reply:
@minhancao : An option for you is to make 'input' RowVector mutable. So we don't change input in-place but rather create a new vector and assign it to input.

Can you try that ?

Also, please create an issue for the childrenSize_ const problem.

Reply:
@aditi-pandit Do you mean to just create a new vector of new batches that would include the k0 and k1 columns in each batch and then re-assign it to input?

I will create an issue for the childrenSize_ const problem.